lib/common/output_log.c

```diff
@@ -23,8 +24,43 @@ typedef struct private_data_s {
     /* gathered in log_begin_list */
     GQueue/*<char*>*/ *prefixes;
     uint8_t log_level;
+    const char *function;
+    const char *file;
+    uint32_t line;
+    uint32_t tags;
 } private_data_t;
 
+/*!
+ * \internal
+ * \brief Log a message using output object's log level and filters
+ *
+ * \param[in] priv    Output object's private_data_t
+ * \param[in] fmt     printf(3)-style format string
+ * \param[in] args... Format string arguments
+ */
+#define logger(priv, fmt, args...) do {                                     \
+        qb_log_from_external_source(pcmk__s((priv)->function, __func__),    \
+            pcmk__s((priv)->file, __FILE__), fmt, (priv)->log_level,        \
+            (((priv)->line == 0)? __LINE__ : (priv)->line), (priv)->tags,   \
+            ##args);                                                        \
+    } while (0);
+
+/*!
+ * \internal
+ * \brief Log a message using an explicit log level and output object's filters
+ *
+ * \param[in] priv    Output object's private_data_t
+ * \param[in] level   Log level
+ * \param[in] fmt     printf(3)-style format string
+ * \param[in] ap      Variadic arguments
+ */
+#define logger_va(priv, level, fmt, ap) do {                                \
+        qb_log_from_external_source_va(pcmk__s((priv)->function, __func__), \
+            pcmk__s((priv)->file, __FILE__), fmt, (level),                  \
```

Comment:
`level` doesn't need parentheses since it's used in an expression by itself (for example, not `(level) * 2`)

Reply:
yep